src/test/java/uk/gov/hmcts/cmc/claimstore/models/offers/SettlementTest.java

```diff
@@ -48,6 +49,82 @@ public void getLastStatementShouldReturnLastStatement() {
         assertThat(settlement.getLastStatement().getOffer().get()).isEqualTo(counterOffer);
     }
 
+    @Test(expected = IllegalSettlementStatementException.class)
+    public void theSamePartyIsNotAllowedToMakeTwoOffersInRow() {
+        settlement.makeOffer(offer, MadeBy.DEFENDANT);
+        settlement.makeOffer(offer, MadeBy.DEFENDANT);
+    }
+
+    @Test
+    public void claimantCanAcceptAnOffer() {
+        settlement.makeOffer(offer, MadeBy.DEFENDANT);
+        settlement.accept(MadeBy.CLAIMANT);
+
+        assertThat(settlement.getLastStatement().getType()).isEqualTo(StatementType.ACCEPTATION);
+    }
+
+    @Test
+    public void defendantCanAcceptAnOffer() {
+        settlement.makeOffer(offer, MadeBy.CLAIMANT);
+        settlement.accept(MadeBy.DEFENDANT);
+
+        assertThat(settlement.getLastStatement().getType()).isEqualTo(StatementType.ACCEPTATION);
+    }
+
+    @Test(expected = IllegalSettlementStatementException.class)
+    public void partyIsNotAllowedToAcceptOfferWhenOfferWasNotMade() {
+        settlement.accept(MadeBy.CLAIMANT);
+    }
+
+    @Test(expected = IllegalSettlementStatementException.class)
+    public void partyIsNotAllowedToAcceptOfferWhenOfferWasAlreadyAcceptedByThemselves() {
+        settlement.makeOffer(offer, MadeBy.DEFENDANT);
+        settlement.accept(MadeBy.CLAIMANT);
+
+        settlement.accept(MadeBy.CLAIMANT);
+    }
+
+    @Test(expected = IllegalSettlementStatementException.class)
+    public void partyIsNotAllowedToAcceptOfferWhenTheOnlyOfferWasMadeByThemselves() {
+        settlement.makeOffer(offer, MadeBy.CLAIMANT);
+        settlement.accept(MadeBy.CLAIMANT);
+    }
+
+    @Test
+    public void claimantCanRejectAnOffer() {
+        settlement.makeOffer(offer, MadeBy.DEFENDANT);
+        settlement.reject(MadeBy.CLAIMANT);
+
+        assertThat(settlement.getLastStatement().getType()).isEqualTo(StatementType.REJECTION);
+    }
+
+    @Test
+    public void defendantCanRejectAnOffer() {
+        settlement.makeOffer(offer, MadeBy.CLAIMANT);
+        settlement.reject(MadeBy.DEFENDANT);
+
+        assertThat(settlement.getLastStatement().getType()).isEqualTo(StatementType.REJECTION);
+    }
+
+    @Test(expected = IllegalSettlementStatementException.class)
+    public void partyIsNotAllowedToRejectOfferWhenOfferWasNotMade() {
+        settlement.reject(MadeBy.CLAIMANT);
+    }
+
+    @Test(expected = IllegalSettlementStatementException.class)
+    public void partyIsNotAllowedToRejectOfferWhenOfferWasAlreadyRejectedByThemselves() {
+        settlement.makeOffer(offer, MadeBy.DEFENDANT);
+        settlement.reject(MadeBy.CLAIMANT);
+
+        settlement.reject(MadeBy.CLAIMANT);
+    }
+
+    @Test(expected = IllegalSettlementStatementException.class)
+    public void partyIsNotAllowedToRejectOfferWhenTheOnlyOfferWasMadeByThemselves() {
```

Comment:
Any party shouldn't be allowed to double accept/reject, not only if it was them doing it.

Reply:
Also -- when an offer has been accepted no further actions should be allowed on that settlement. We should have test cases for that.

Reply:
`partyIsNotAllowedToRejectOfferWhenTheOnlyOfferWasMade`